### PR TITLE
Update changelog for 4.8.0 release.

### PR DESCRIPTION
**Pull request by mdickinson**
This PR adds changelog entries for the upcoming 4.8.0 release. I'll make a release branch once this is merged.

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -2,6 +2,89 @@
...
+  (#173)
+- Remove the non-functional ``RefreshCodePlugin``. (#202)
+- Remove ``project_runnable``, which was never functional. (#169)
+- Remove outdated debugging fallback from the ``ExtensionPoint`` source. (#167)
+- Remove ``FBIPlugin``. (#166)
+- Remove the ``remote_editor`` plugins. (#137)
+
+Documentation
+-------------
+- Add docstrings for tasks plugin extension points. (#181)
```
Comment (mdickinson) [inline CHANGES.txt]: Consistency nitpick: missing blank line.

File: CHANGES.txt [hunk 2]
```diff
@@ -2,6 +2,89 @@
...
+- Use a fixed pickle protocol when saving task layout state, to avoid
+  cross-Python-version difficulties. (#179)
+- Fix deprecation warnings from use of ``Logger.warn``. (#178)
+- Fix some Python 3 syntax errors in example scripts. (#171)
+
+Removals
+--------
+
+- Remove the unsupported and incomplete ``UpdateCheckerPlugin``. (#199)
+- Remove the plugin.debug empty submodule. (#195)
```
Comment (mdickinson) [inline CHANGES.txt]: nitpick: add backticks around ``plugin.debug``

File: CHANGES.txt [hunk 3]
```diff
@@ -2,6 +2,89 @@
...
+- Remove the old IPythonShell plugin, which was based on pre-IPython 1.0.
```
Comment (mdickinson) [inline CHANGES.txt]: nitpick: backticks

File: CHANGES.txt [hunk 4]
```diff
@@ -2,6 +2,89 @@
...
+  (#173)
+- Remove the non-functional ``RefreshCodePlugin``. (#202)
+- Remove ``project_runnable``, which was never functional. (#169)
+- Remove outdated debugging fallback from the ``ExtensionPoint`` source. (#167)
+- Remove ``FBIPlugin``. (#166)
+- Remove the ``remote_editor`` plugins. (#137)
```
Comment (rahulporuri) [inline CHANGES.txt]: nit : plugin, not plugins.

File: CHANGES.txt [hunk 5]
```diff
@@ -2,6 +2,89 @@
...
+
+Documentation
+-------------
+- Add docstrings for tasks plugin extension points. (#181)
+- Fix incorrect documentation for always_use_default_layout. (#177)
```
Comment (mdickinson) [inline CHANGES.txt]: backticks

File: CHANGES.txt (hunk 4 above)
Comment (rahulporuri) [reply CHANGES.txt]: (or is it actually multiple plugins?)

Comment (mdickinson) [reply CHANGES.txt]: It's actually multiple things; there isn't a single `RemoteEditorPlugin`.

**Pull request merged**